tsl/test/expected/hypercore_index_btree.out

```diff
@@ -1062,3 +1062,115 @@ select * from only_nulls_null;
 ---------+------+--------+----------+-------+------------
 (0 rows)
 
+--------------------------------------------------
+-- Test unique index creation                   --
+--------------------------------------------------
+create table uniquetable (time timestamptz not null, value int);
+select create_hypertable('uniquetable', 'time', create_default_indexes => false);
+    create_hypertable     
+--------------------------
+ (7,public,uniquetable,t)
+(1 row)
+
+insert into uniquetable values ('2024-01-01 01:00', 1), ('2024-01-01 02:00', 2);
+select format('%I.%I', chunk_schema, chunk_name)::regclass as unique_chunk
+  from timescaledb_information.chunks
+ where format('%I.%I', hypertable_schema, hypertable_name)::regclass = 'uniquetable'::regclass
+ order by unique_chunk asc
+ limit 1 \gset
+alter table uniquetable set (timescaledb.compress_orderby='time');
+WARNING:  there was some uncertainty picking the default segment by for the hypertable: You do not have any indexes on columns that can be used for segment_by and thus we are not using segment_by for compression. Please make sure you are not missing any indexes
+NOTICE:  default segment by for hypertable "uniquetable" is set to ""
+-- Create a non-Hypercore TAM compressed chunk
+select * from compress_chunk(:'unique_chunk');
+             compress_chunk              
+-----------------------------------------
+ _timescaledb_internal._hyper_7_19_chunk
+(1 row)
+
+-- Should still be a "heap" chunk
+select c.relname, a.amname from pg_class c
+inner join pg_am a ON (c.relam = a.oid)
+where c.oid = :'unique_chunk'::regclass;
+      relname      | amname 
+-------------------+--------
+ _hyper_7_19_chunk | heap
+(1 row)
+
+insert into uniquetable values ('2024-01-01 01:00', 3);
+-- Unique index creation on compressed chunk not supported
+\set ON_ERROR_STOP 0
+create unique index time_key on uniquetable (time);
+ERROR:  operation not supported on hypertables that have compression enabled
+\set ON_ERROR_STOP 1
```

Comment:
Although it does not harm to have this check here, I am not sure what potential bug you wanted to catch here.

Reply:
Just to make clear there is a difference and that this error is intact despite supporting unique indexes on Hypercore TAM.